### PR TITLE
Jordyro1/oracle module library

### PR DESCRIPTION
**Pull request by JordyRo1**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix
X Feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
- Documentation content changes
- Testing
- Other (please describe):

## What is the current behavior?

Resolves: #297 

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

 No

### CODE REVIEW

File: src/oracle/oracle_modules.cairo [hunk 1]
```diff
@@ -10,8 +10,11 @@
 use satoru::data::data_store::{IDataStoreDispatcher, IDataStoreDispatcherTrait};
 use satoru::event::event_emitter::{IEventEmitterDispatcher, IEventEmitterDispatcherTrait};
 use satoru::oracle::{
-    oracle::IOracleDispatcher, oracle_utils::{SetPricesParams, SimulatePricesParams},
+    oracle::{IOracleDispatcher, IOracleDispatcherTrait}, oracle_utils::{SetPricesParams, SimulatePricesParams},
 };
+use satoru::price::price::Price;
+use satoru::oracle::error::OracleError;
+use starknet::ContractAddress;
```
Comment (zarboq) [inline src/oracle/oracle_modules.cairo:17]: Add this under `// Core lib imports`

**Review by zarboq — CHANGES_REQUESTED** [review]
Minor change then good to go !

**Comment by zarboq** [comment]
Fix checks also please

File: src/oracle/oracle_modules.cairo [hunk 2]
```diff
@@ -5,16 +5,15 @@
 //                                  IMPORTS
 // *************************************************************************
 // Core lib imports.
-
+use satoru::price::price::Price;
```
Comment (zarboq) [inline src/oracle/oracle_modules.cairo:8]: I meant only starknet::* imports under `Core lib`

satoru::* imports always under `Local imports`

Comment (JordyRo1) [reply src/oracle/oracle_modules.cairo:8]: ooh I see

**Review by zarboq — APPROVED** [review]

**Comment by JordyRo1** [comment]
There are still an issue with the test/check. Previously, since there was an empty ```with_oracle_prices_before```, the test expected to fail at invalid_withdraw_key. Now that there is the function ```oracle.set_prices``` it seems to fails at the function ```validate_role``` of the function ```only_controller``` called in ```set_prices```. Is that a normal behaviour or should I dig to find the issue ?

**Comment by zarboq** [comment]
I think you should grant the corresponding role in tests

**Comment by JordyRo1** [comment]
I granted the role::CONTROLLER to the order_keeper (he already had role::ORDER_KEEPER). The issue is that function ```set_prices``` still returns the same error. The issue is that the function ```get_caller_address``` returns an address different from the defined ```order_keeper``` defined in the example.

**Comment by JordyRo1** [comment]
Everything should be good !

**Pull request merged**